app/services/catalog/hierarchical_model_search.rb

```diff
@@ -0,0 +1,50 @@
+module Catalog
```

Comment:
I think this should be moved from this PR and moved to its own.

Reply:
Good idea, I'll pull it out and put a new one in.